metron-platform/metron-common/src/main/java/org/apache/metron/common/utils/KafkaUtils.java

```diff
@@ -68,12 +68,14 @@
     return ret;
   }
 
-  public List<String> fromEndpoint(String url) throws URISyntaxException {
+  public List<String> fromEndpoint(String url){
     List<String> ret = new ArrayList<>();
     if(url != null) {
-      URI uri = new URI(url);
-      int port = uri.getPort();
-      ret.add(uri.getHost() + ((port > 0)?(":" + port):""));
+      Iterable<String> splits = Splitter.on("//").split(url);
+      if(Iterables.size(splits) == 2) {
+        String hostPort = Iterables.getLast(splits);
+        ret.add(hostPort);
```

Comment:
The problem is that the URI based solution that was implemented in that patch (I believe based on my suggestion, sigh) doesn't allow for protocols with underscore characters.  It's entirely independent of SASL_PLAINTEXT, other than the fact that underscore is in it.

The intention was to exactly reverse the chunk you mention, because the original didn't care about underscores.

On further inspection (which I'm glad you did!), these are not equivalent before and after.  Which begs the question of why things worked in testing.  I'll dig in a bit more to figure out what's going on under the hood here.